### ISSUE TITLE
Manual upload the mapping file

### ISSUE THREAD

**Issue by pjq** [firebase/firebase-android-sdk#2004]
## What feature would you like to see?
The Firebase Crashlytics will auto upload the mapping file during the build phase, and I want to have the capability to upload it manually.

## How would you use it?
Our build server will not allow to visit the external url, only allow to the local server, so it can't send the mapping file to the Google Firebase Server.
If it will have the tools/tasks to support manual upload that will be really helpful.

**Comment by tobecrazy** [comment]
same here

**Comment by pjq** [comment]
Will this issue be take care of by Google, or just let it go?
It will be appreciated that if someone can update the status or comment about it.

**Comment by zhangao0086** [comment]
Any updates on this?

**Comment by ChinaZeng** [comment]
Manual upload is necessary. Is there any progress?

**Comment by pjq** [comment]
Will this requirement valid or invalid?

**Comment by ding33211** [comment]
anyone help?

**Comment by mrichards** [comment]
Hey folks we have are aware this is a desired feature and will be supporting it eventually. I don't have a timeline for it yet, though. I'll keep the issue open and we'll post an update here when we have one.

**Comment by pjq** [comment]
It's probably low priority from Google's perspective, but it's one blocker issue for our team, so do you have any workaround for it, or any tricky things can do it?
Or any sample code I can refer if I want to build the "manual upload tool"? Really appreciate someone from the Firebase Android SDK team can give any hint.

**Comment by pjq** [comment]
`
<html>
<body>


  | PUT /v1/project/-/app/1:559704843700:android:65c5bf32ff0b206a25bee1/upload/java/35387219eef244db825e995a8085f9f5 HTTP/1.1
-- | --
User-Agent | crashlytics-gradle/2.6.1
X-CRASHLYTICS-API-CLIENT-TYPE | crashlytics-gradle
X-CRASHLYTICS-API-CLIENT-VERSION | 2.6.1
Content-Length | 5208094
Host | firebasecrashlyticssymbols.googleapis.com
Connection | Keep-Alive
Accept-Encoding | gzip,deflate


</body>
</html>`

**Comment by pjq** [comment]
This is the API called upload the mapping file, so probably we can call it and build the upload tool by ourself.

**Comment by pjq** [comment]
![image](https://user-images.githubusercontent.com/87846/127995910-233e6f2a-18a9-4894-a5ed-af5a4daadb72.png)

**Comment by pjq** [comment]
![image](https://user-images.githubusercontent.com/87846/129133522-e5af2983-7968-493b-abe4-1dfe30e59c09.png)

**Comment by adef145** [comment]
Hi there, any update for this?

My team follow this article https://medium.com/nerd-for-tech/secure-your-firebases-google-services-json-file-in-android-16680f8e5fb4 to secure our google keys, but somehow our crash report still obfuscate.

Anyone familiar with this?

**Comment by ChinaZeng** [comment]
Is there any progress on this?

**Comment by ptsekov** [comment]
Hello,

Does Firebase Console Crashlytics have any means to associate the mappings file uploaded during the build step to a specific app version? If not does it mean that uploading a new mappings file would break deobfuscation of stack traces generated by earlier versions?

Google Play provides the ability to upload a mapping file for a given app version which seems the proper thing to do.

**Comment by mrichards** [comment]
@ptsekov The mapping file is associated with a UUID that is added to the app as a resource during compilation, not the app version code. So unless you've hardcoded that UUID, new builds will not overwrite old ones, even if the version code is the same.

As a general update, we have prioritized work on the manual (non-Gradle) mapping file upload utility. We don't have an ETA for release yet but it is being worked on.

**Comment by SirGoingFar** [comment]
> Hi there, any update for this?
> 
> My team follow this article https://medium.com/nerd-for-tech/secure-your-firebases-google-services-json-file-in-android-16680f8e5fb4 to secure our google keys, but somehow our crash report still obfuscate.
> 
> Anyone familiar with this?

@adef145 I did something similar for my team. Do you have a workaround yet?

**Comment by adef145** [comment]
> > Hi there, any update for this?
> > My team follow this article https://medium.com/nerd-for-tech/secure-your-firebases-google-services-json-file-in-android-16680f8e5fb4 to secure our google keys, but somehow our crash report still obfuscate.
> > Anyone familiar with this?
> 
> @adef145 I did something similar for my team. Do you have a workaround yet?

Currently we ignore it and in parallel trying Sentry instead

**Comment by mrichards** [comment]
Hi all, the capability to upload mapping files outside of Gradle is now available in the [Firebase CLI](https://firebase.google.com/docs/cli#crashlytics-commands). If you have any issues with this feature, please report them in the [firebase-tools repo](https://github.com/firebase/firebase-tools). Thank you!

**Comment by adef145** [comment]
> Hi all, the capability to upload mapping files outside of Gradle is now available in the [Firebase CLI](https://firebase.google.com/docs/cli#crashlytics-commands). If you have any issues with this feature, please report them in the [firebase-tools repo](https://github.com/firebase/firebase-tools). Thank you!

Thanks @mrichards dan teams
